### DISCUSSION TITLE
Preferences for Inlay Hints

### DISCUSSION THREAD

**Discussion by ahmedneilhussain** [eclipse/lsp4e#687] [Ideas]
AFAICT inlay hints in lsp4e are always on if the server supports them - is that correct?

We like them but some users aren't enamoured. They can be turned off for Java IIRC but that's a totally separate plugin/editor, there's nothing that can control them for LSP4e-enabled editors.

If so, would it be feasible to add preferences to control them, and if so what would be the right level to implement it?

Thoughts welcome...

**Comment by rubenporras** [comment]
Hi @ahmedneilhussain ,

I think that if some users like it and some not, there is no way around having configuration options visible to the user. See for example the discussion in https://github.com/eclipse-platform/eclipse.platform.text/issues/211, it could make sense to have a common configuration page for several of this features which users might like/dislike.

You could check VS Code as I did, but if the JDT offers a configuration option, I would guess that VSCode does it as well.
